lightning/src/offers/flow.rs

```diff
@@ -0,0 +1,1113 @@
+		ES::Target: EntropySource,
+	{
+		let node_id = self.get_our_node_id();
+		let expanded_key = &self.inbound_payment_key;
+		let entropy = &*entropy_source;
+		let secp_ctx = &self.secp_ctx;
+
+		let nonce = Nonce::from_entropy_source(entropy);
+		let context = OffersContext::OutboundPayment { payment_id, nonce, hmac: None };
+
+		let path = self
+			.create_blinded_paths_using_absolute_expiry(context, Some(absolute_expiry), peers)
+			.and_then(|paths| paths.into_iter().next().ok_or(()))
+			.map_err(|_| Bolt12SemanticError::MissingPaths)?;
+
+		let builder = RefundBuilder::deriving_signing_pubkey(
+			node_id,
+			expanded_key,
+			nonce,
+			secp_ctx,
+			amount_msats,
+			payment_id,
+		)?
+		.chain_hash(self.chain_hash)
+		.absolute_expiry(absolute_expiry)
+		.path(path);
+
+		Ok(builder)
+	}
+
+	/// Creates an [`InvoiceRequestBuilder`] such that the [`InvoiceRequest`] it builds is recognized
+	/// by the [`OffersMessageFlow`], and any corresponding [`Bolt12Invoice`] received in response
+	/// can be verified using [`Self::verify_bolt12_invoice`].
+	///
+	/// # Nonce
+	/// The nonce is used to create a unique [`InvoiceRequest::payer_metadata`] for the invoice request.
+	/// These will be used to verify the corresponding [`Bolt12Invoice`] when it is received.
+	///
+	/// [`ChannelManager::send_payment`]: crate::ln::channelmanager::ChannelManager::send_payment
+	/// [`ChannelManager`]: crate::ln::channelmanager::ChannelManager
+	pub fn create_invoice_request_builder<'a>(
+		&'a self, offer: &'a Offer, nonce: Nonce, quantity: Option<u64>, amount_msats: Option<u64>,
+		payer_note: Option<String>, human_readable_name: Option<HumanReadableName>,
+		payment_id: PaymentId,
+	) -> Result<InvoiceRequestBuilder<'a, 'a, secp256k1::All>, Bolt12SemanticError> {
+		let expanded_key = &self.inbound_payment_key;
+		let secp_ctx = &self.secp_ctx;
+
+		let builder: InvoiceRequestBuilder<secp256k1::All> =
+			offer.request_invoice(expanded_key, nonce, secp_ctx, payment_id)?.into();
+		let builder = builder.chain_hash(self.chain_hash)?;
+
+		let builder = match quantity {
+			None => builder,
+			Some(quantity) => builder.quantity(quantity)?,
+		};
+		let builder = match amount_msats {
+			None => builder,
+			Some(amount_msats) => builder.amount_msats(amount_msats)?,
+		};
+		let builder = match payer_note {
+			None => builder,
+			Some(payer_note) => builder.payer_note(payer_note),
+		};
+		let builder = match human_readable_name {
+			None => builder,
+			Some(hrn) => builder.sourced_from_human_readable_name(hrn),
+		};
+
+		Ok(builder)
+	}
+
+	/// Creates a [`StaticInvoiceBuilder`] from the corresponding [`Offer`] and [`Nonce`] that were
+	/// created via [`Self::create_async_receive_offer_builder`].
+	#[cfg(async_payments)]
+	pub fn create_static_invoice_builder<'a, ES: Deref, R: Deref>(
+		&self, router: &R, entropy_source: ES, offer: &'a Offer, offer_nonce: Nonce,
+		amount_msats: Option<u64>, payment_secret: PaymentSecret, relative_expiry_secs: u32,
```

Comment:
Static invoices are amount-less so no need for this argument

Reply:
Thanks a lot for the clarification! I fixed the function signature to remove amount_msats in [pr3639.17](https://github.com/shaavan/rust-lightning/commits/pr3639.17)
Thanks!